tools/shared/src/test/scala/org/scalajs/core/tools/linker/AnalyzerTest.scala

```diff
@@ -0,0 +1,240 @@
+        reqsFactory.callMethod("LA", "foo__V"))
+
+    assertHasError("MissingMethod(LA.foo__V)", analysis) {
+      case MissingMethod(foo, `fromUnitTest`) =>
+        foo.owner.encodedName == "LA" && foo.encodedName == "foo__V"
+    }
+  }
+
+  @Test
+  def conflictingDefaultMethods(): Unit = {
+    val defaultMethodDef = MethodDef(static = false, Ident("foo__V"), Nil,
+        NoType, Some(Skip()))(emptyOptHints, None)
+    val classDefs = Seq(
+        ClassDef(Ident("LI1"), ClassKind.Interface, None, Nil,
+            None, List(defaultMethodDef), Nil)(emptyOptHints),
+        ClassDef(Ident("LI2"), ClassKind.Interface, None, Nil,
+            None, List(defaultMethodDef), Nil)(emptyOptHints),
+        ClassDef(Ident("LA"), ClassKind.Class, Some(Ident(ObjectClass)),
+            List(Ident("LI1"), Ident("LI2")), None, List(trivialCtor("LA")),
+            Nil)(emptyOptHints)
+    )
+
+    val analysis = computeAnalysis(classDefs,
+        reqsFactory.instantiateClass("LA", "init___") ++
+        reqsFactory.callMethod("LA", "foo__V"))
+
+    assertHasError("ConflictingDefaultMethods(LI1.foo__V, LI2.foo__V)", analysis) {
+      case ConflictingDefaultMethods(List(a, b), `fromAnalyzer`) =>
+        a.owner.encodedName == "LI1" && a.encodedName == "foo__V" &&
+        b.owner.encodedName == "LI2" && b.encodedName == "foo__V"
+    }
+  }
+
+  private def computeAnalysis(classDefs: Seq[ClassDef],
+      symbolRequirements: SymbolRequirement = reqsFactory.none(),
+      stdlib: Boolean = true): Analysis = {
+
+    val classesWithEntryPoints0 = classDefs
+      .map(ir.EntryPointsInfo.forClassDef)
+      .withFilter(_.hasEntryPoint)
+      .map(_.encodedName)
+
+    val encodedNameToInfo =
+      classDefs.map(c => c.name.name -> Infos.generateClassInfo(c)).toMap
+
+    val inputProvider = new Analyzer.InputProvider {
+      def classesWithEntryPoints(): TraversableOnce[String] =
+        classesWithEntryPoints0
+
+      def loadInfo(encodedName: String): Option[Infos.ClassInfo] = {
+        val own = encodedNameToInfo.get(encodedName)
+        if (stdlib) own.orElse(TestIRRepo.loadInfo(encodedName))
+        else own
+      }
+    }
+
+    Analyzer.computeReachability(CommonPhaseConfig(), symbolRequirements,
+        allowAddingSyntheticMethods = true, inputProvider)
+  }
+
+  private def assertNoError(analysis: Analysis): Unit =
+    assertHasErrors(analysis)
+
+  private def assertHasErrors(analysis: Analysis,
+      expectedErrors: Error*): Unit = {
+    val actualErrors = analysis.errors
+
+    for (expectedError <- expectedErrors) {
+      assertTrue(s"Missing expected error: $expectedError",
+          actualErrors.contains(expectedError))
+    }
+
+    if (actualErrors.size != expectedErrors.size) {
+      for (actualError <- actualErrors) {
+        assertTrue(s"Unexpected error: $actualError",
+            expectedErrors.contains(actualError))
+      }
+    }
+  }
+
+  private def assertHasError(msg: String, analysis: Analysis)(
```

Comment:
Consider calling this `assertContainsError` to make it more explicit that it need not be exhaustive.